### PR TITLE
WIP - Chore/useryaml

### LINKED ISSUE #188: create user list syncing job
create a k8s job that runs every hour to grab user lists from kids first s3 bucket and sync to fence db
- need to get s3 path from kids first
- need to agree on how to grant the k8s job access to the file

### LINKED ISSUE #208: isolate revproxy csrf check to core services
the csrf header-to-cookie check in the revproxy currently covers every endpoint.
Let's restrict that to indexd, sheepdog, peregrine, and fence (except for the approval form POST endpoint), so we don't break things like jupyter and the fence oauth approval form.

### PR DESCRIPTION
**Pull request by frickjack**
resolves #188 
resolves #208 

You can test interactive in the `planxplanetv1` VPC:
      https://github.com/uc-cdis/cdis-wiki/blob/master/ops/AWS-Accounts.md

```
$ ssh k8s.planx-pla.net
$ g3k help

# run fence-create against current fence configmap
$ g3k runjob useryaml     
$ g3k joblogs useryaml

# copy useryaml from S3 path in gobal configmap, update fence configmap, run fence-create
$ g3k runjob useryamls3 
$ g3k joblogs useryamls3

# there's a useryamls3 cron-job currently scheduled to run @hourly
$ kubectl get cronjobs
```
* augment kubes.sh to support accessing helper functions via a master 'g3k' function (short for *gen3 k8s*):
```
$ g3k help
  
  Use:
  g3k COMMAND - where COMMAND is one of:
    help
    jobpods JOBNAME - list pods associated with given job
    joblogs JOBNAME - get logs from first result of jobpods
    pod PATTERN - grep for the first pod name matching PATTERN
    pods PATTERN - grep for all pod names matching PATTERN
    replicas DEPLOYMENT-NAME REPLICA-COUNT
    roll DEPLOYMENT-NAME
      Apply a superfulous metadata change to a deployment to trigger
      the deployment's running pods to update
    runjob JOBNAME 
     - JOBNAME also maps to cloud-automation/kube/services/JOBNAME-job.yaml
  update_config CONFIGMAP-NAME YAML-FILE
```

* setup `useryaml` and `useryamls3` k8s jobs, and a `useryamls3` cronjob - here is a copy of the README (https://github.com/uc-cdis/cloud-automation/blob/chore/useryaml/kube/services/jobs/README.md):

# TL;DR

K8s jobs to do various things.  The `g3k` bash helper functions (from `kube.sh`) provide an easy way to run a job - ex: `g3k runjob useryaml`

## Setup

The `cloud-automation/tf_files/configs/kube-setup-roles.sh` scripts sets up the k8s roles
required by these jobs.  It runs automatically as part of the `kube-services` script that
boots up a commons, but may need to be run manually to setup existing commons to run jobs.

## useryamls3-cronjob

Periodically syncs a user.yaml file from a specified S3 buket into the k8s `fence` configmap,
and update fence's user-access database.

Note that this job assumes that the k8s worker nodes have an IAM policy that allows S3 read -
which is the default in new commons deployments, but may require a manual update to existing commons.

## useryamls3-job

Syncs a user.yaml file from the S3 bucket specified in the global configmap's `useryaml_s3path` attribute (`kubectl get configmap global -o=jsonpath='{.data.useryaml_s3path}'`) into the k8s `fence` configmap,
and update fence's user-access database.

Note that this job assumes that the k8s worker nodes have an IAM policy that allows S3 read -
which is the default in new commons deployments, but may require a manual update to existing commons.

## useryaml-job

Sync the `user.yaml` from the k8s `fence` configmap into fence's database.  A typical workflow would first update the fence configmap, then run this job - ex:
```
$ g3k update_config fence apis_configs/user.yaml
$ g3k runjob useryaml
```

### CODE REVIEW

File: Docker/awshelper/push.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+docker tag cdis-awshelper:1.0.0 quay.io/cdis/awshelper:1.0.0
```
Comment (philloooo) [inline Docker/awshelper/push.sh]: remove

File: kube/README.md [hunk 2]
```diff
@@ -236,15 +237,49 @@ Kafka cluster that support different data streams for the system.
...
 
-### bash functions
+# bash functions
+
 There are some helper functions in [kubes.sh](https://github.com/uc-cdis/cloud-automation/blob/master/kube/kubes.sh) for k8s related operations.
 
-#### patch_kube
-`patch_kube $DEPLOYMENT_NAME` to let k8s recycle pods when there is no change.
+### patch_kube
+`g3k patch_kube $DEPLOYMENT_NAME` to let k8s recycle pods when there is no change.
```
Comment (philloooo) [inline kube/README.md]: `g3k roll $DEPLOYMENT_NAME` now?
